File: uri/1021/1021.cpp
#include <cstdio>

int main(){
    double n,  d[] = {100.0, 50.0, 20.0, 10.0, 5.0, 2.0, 1.0, 0.5, 0.25, 0.10, 0.05, 0.01};
    int t = 0, c, i;

    scanf("%lf", &n);
    printf("NOTAS:\n");

    t = 0;
    for (i = 0; i < 12; i++) {
        c = 0;
        while (n >= d[t]){
            n -= d[t];
            c++;
        }
        if (d[t] == 1.0)
            printf("MOEDAS:\n");
        if (d[t] >= 2.0 )
            printf("%d nota(s) de R$ %.2f\n", c, d[t]);
        else
            printf("%d moeda(s) de R$ %.2f\n", c, d[t]);
        t++;
    }
    return 0;
}

File: solutions/beecrowd/1021/1021.clj
(ns main
  (:require [clojure.string :as string]))

(def notas [10000.0 5000.0 2000.0 1000.0 500.0 200.0])
(def coins [100.0 50.0 25.0 10.0 5.0 1.0])

(defn print-result [value currencies currency-type]
  (println (str (string/upper-case currency-type) "S:"))
  (loop [value value
         [currency & remaining-currencies] currencies]
    (let [quantity (quot value currency)]
      (printf
       "%.0f %s(s) de R$ %.2f%n"
       quantity
       currency-type
       (with-precision 2 (/ currency 100.0))))
    (if (seq remaining-currencies)
      (recur (mod value currency) remaining-currencies)
      (mod value currency))))

(defn main []
  (loop [line (read-line)]
    (when line
      (let [value (-> line
                      Double/parseDouble)]
        (-> (* value 100)
            (print-result notas "nota")
            (print-result coins "moeda"))
        (recur (read-line))))))

(main)
